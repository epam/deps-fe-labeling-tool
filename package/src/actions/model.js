
import { batch } from 'react-redux'
import { createAction } from 'redux-actions'
import { LabelType, LABEL_TYPE_NAME } from '@/models/Label'
import { Markup } from '@/models/Markup'
import { markupSelector } from '@/selectors/markup'
import { fieldsSelector, fieldsToDeleteSelector } from '@/selectors/model'
import {
  updateLabelsWithSettings,
  updateTablesWithSettings
} from './markup'

const FEATURE_NAME = 'MODEL'

const storeFields = createAction(
  `${FEATURE_NAME}/STORE`
)

const storeInitialFields = createAction(
  `${FEATURE_NAME}/STORE_INITIAL_FIELDS`
)

const updateInitialFields = createAction(
  `${FEATURE_NAME}/UPDATE_INITIAL_FIELDS`
)

const enableDeleteFieldsMode = createAction(
  `${FEATURE_NAME}/ENABLE_DELETE_FIELD_MODE`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

const updateFieldsToDelete = createAction(
  `${FEATURE_NAME}/UPDATE_FIELDS_TO_DELETE`
)

const cancelDeleteFieldsMode = createAction(
  `${FEATURE_NAME}/CANCEL_DELETE_FIELDS_MODE`
)

const confirmFieldsDeletion = () => (dispatch, getState) => {
  const state = getState()
  const markup = markupSelector(state)
  const fields = fieldsSelector(state)
  const fieldsToDelete = fieldsToDeleteSelector(state)
  const newFields = fields.filter((field) => fieldsToDelete.every((f) => f.code !== field.code))
  const assignedObjects = markup
    ? Markup.getAllObjects(markup).filter((markupObject) => fieldsToDelete.find((field) => field.code === markupObject.fieldCode))
    : []

  batch(() => {
    assignedObjects.forEach((markupObject) => {
      if (markupObject.typeName === LABEL_TYPE_NAME) {
        dispatch(updateLabelsWithSettings({
          ...markupObject,
          fieldCode: '',
          type: LabelType.UNASSIGNED,
          index: undefined
        }))
      } else {
        dispatch(updateTablesWithSettings({
          ...markupObject,
          fieldCode: '',
          index: undefined
        }))
      }
    })
    dispatch(storeFields(newFields))
    dispatch(cancelDeleteFieldsMode())
  })
}

export {
  enableDeleteFieldsMode,
  resetDefault,
  storeFields,
  storeInitialFields,
  updateFieldsToDelete,
  updateInitialFields,
  confirmFieldsDeletion,
  cancelDeleteFieldsMode
}
