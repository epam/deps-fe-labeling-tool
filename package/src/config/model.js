import { FieldType } from '@/enums/FieldType'
import { Field } from '@/models/Field'
import { PairFieldMeta, ListFieldMeta, EnumFieldMeta } from '@/models/FieldMeta'

const fields = [
  new Field(
    'dates',
    'Dates',
    FieldType.DATE
  ),
  new Field(
    'firstName',
    'First Name',
    FieldType.STRING
  ),
  new Field(
    'lastName',
    'Last Name',
    FieldType.STRING,
    undefined,
    true
  ),
  new Field(
    'companyName',
    'Company Name',
    FieldType.STRING,
    undefined,
    true
  ),
  new Field(
    'city',
    'City',
    FieldType.ENUM,
    new EnumFieldMeta([
      'Anytown',
      'Sometown',
      'San Diego',
      'Los Angeles'
    ]),
    true
  ),
  new Field(
    'countryStates',
    'Country States',
    FieldType.LIST,
    new ListFieldMeta(FieldType.STRING),
    true
  ),
  new Field(
    'dateMatching',
    'Date Matching',
    FieldType.LIST,
    new ListFieldMeta(FieldType.CHECKMARK),
    true
  ),
  new Field(
    'date',
    'Date',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING)
  ),
  new Field(
    'zipCodes',
    'Zip Codes',
    FieldType.LIST,
    new ListFieldMeta(
      FieldType.PAIR,
      new PairFieldMeta(FieldType.STRING, FieldType.STRING)
    ),
    true
  ),
  new Field(
    'homeAddress',
    'Home Address',
    FieldType.STRING
  ),
  new Field(
    'businessAddress',
    'Business Address',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING),
    true
  ),
  new Field(
    'taxpayerSpousesInfo',
    'Taxpayer Spouse\'s Info',
    FieldType.TABLE
  ),
  new Field(
    'taxpayerInfo',
    'Taxpayer Info',
    FieldType.TABLE,
    undefined,
    true
  ),
  new Field(
    'tables',
    'Tables',
    FieldType.LIST,
    new ListFieldMeta(FieldType.TABLE)
  ),
  new Field(
    'uniqueIdentifier',
    'Unique Identifier',
    FieldType.PAIR,
    new PairFieldMeta(FieldType.STRING, FieldType.STRING)
  ),
  new Field(
    'signatory',
    'Signatory',
    FieldType.CHECKMARK
  ),
  new Field(
    'returnTranscript',
    'Return Transcript',
    FieldType.CHECKMARK
  ),
  new Field(
    'formNumbers',
    'Form Numbers',
    FieldType.LIST,
    new ListFieldMeta(
      FieldType.ENUM,
      new EnumFieldMeta([
        '1098',
        '5498',
        '1040'
      ])
    )
  )
]

export {
  fields
}
