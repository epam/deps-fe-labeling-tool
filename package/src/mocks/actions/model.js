const storeFields = jest.fn((fields) => ({
  type: 'MOCK_STORE_FIELDS_ACTION',
  payload: fields
}))

const updateFieldsToDelete = jest.fn((fields) => ({
  type: 'MOCK_UPDATE_FIELDS_TO_DELETE_ACTION',
  payload: fields
}))

const storeInitialFields = jest.fn((fields) => ({
  type: 'MOCK_STORE_INITIAL_FIELDS_ACTION',
  payload: fields
}))

const mockModelActions = {
  storeFields,
  storeInitialFields,
  updateFieldsToDelete
}

export {
  mockModelActions
}
