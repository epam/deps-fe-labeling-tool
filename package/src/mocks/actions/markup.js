const addLabels = jest.fn((labels) => ({
  type: 'MOCK_ADD_LABELS_ACTION',
  payload: labels
}))

const addTables = jest.fn((tables) => ({
  type: 'MOCK_ADD_TABLES_ACTION',
  payload: tables
}))

const removeLabels = jest.fn((labels) => ({
  type: 'MOCK_REMOVE_LABELS_ACTION',
  payload: labels
}))

const updateLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_LABELS_ACTION',
  payload: labels
}))

const updateAllLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_ALL_LABELS_ACTION',
  payload: labels
}))

const updateTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_TABLES_ACTION',
  payload: tables
}))

const updateAllTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_ALL_TABLES_ACTION',
  payload: tables
}))

const selectLabels = jest.fn((labels) => ({
  type: 'MOCK_SELECT_LABELS_ACTION',
  payload: labels
}))

const selectTables = jest.fn((tables) => ({
  type: 'MOCK_SELECT_TABLES_ACTION',
  payload: tables
}))

const clearSelection = jest.fn(() => ({
  type: 'MOCK_CLEAR_SELECTION_ACTION'
}))

const removeTables = jest.fn((tables) => ({
  type: 'MOCK_REMOVE_TABLES_ACTION',
  payload: tables
}))

const importMarkup = jest.fn(() => ({
  type: 'MOCK_IMPORT_MARKUP'
}))

const exportMarkup = jest.fn(() => ({
  type: 'MOCK_EXPORT_MARKUP'
}))

const undo = jest.fn(() => ({
  type: 'MOCK_UNDO_ACTION'
}))

const redo = jest.fn(() => ({
  type: 'MOCK_REDO_ACTION'
}))

const updateInitialMarkup = jest.fn(() => ({
  type: 'MOCK_UPDATE_INITIAL_MARKUP_ACTION'
}))

const storeMarkup = jest.fn((markup) => ({
  type: 'MOCK_STORE_MARKUP_ACTION',
  payload: markup
}))

const storeAssignedMarkup = jest.fn((markup) => ({
  type: 'MOCK_UPDATE_MARKUP_ACTION',
  payload: markup
}))

const updatePageMarkup = jest.fn(() => ({
  type: 'UPDATE_PAGE_MARKUP'
}))

const reset = jest.fn(() => ({
  type: 'RESET_MARKUP_ACTION'
}))

const copyMarkup = jest.fn(() => ({
  type: 'COPY_MARKUP_ACTION'
}))

const pasteMarkup = jest.fn(() => ({
  type: 'PASTE_MARKUP_ACTION'
}))

const updateLabelsWithSettings = jest.fn(() => ({
  type: 'UPDATE_LABELS_WITH_SETTINGS'
}))

const updateTablesWithSettings = jest.fn(() => ({
  type: 'UPDATE_TABLES_WITH_SETTINGS'
}))

const mockMarkupActions = {
  addLabels,
  addTables,
  copyMarkup,
  pasteMarkup,
  removeLabels,
  reset,
  updateLabels,
  updateAllLabels,
  updateTables,
  updateAllTables,
  selectLabels,
  selectTables,
  clearSelection,
  removeTables,
  importMarkup,
  exportMarkup,
  undo,
  updateInitialMarkup,
  storeMarkup,
  storeAssignedMarkup,
  updatePageMarkup,
  redo,
  updateLabelsWithSettings,
  updateTablesWithSettings
}

export {
  mockMarkupActions
}
