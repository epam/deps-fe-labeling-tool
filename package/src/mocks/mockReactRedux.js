const mockDispatch = jest.fn((action) => action)

const createMockMapStateToProps = (mapStateToProps) => {
  if (!mapStateToProps) {
    return mapStateToProps
  }

  return (state, ownProps) => ({
    props: mapStateToProps(state, ownProps)
  })
}

const getDispatchFunctionFormShorthandObject = (actionsObj) => (dispatch) => (
  Object.keys(actionsObj).reduce(
    (props, actionKey) => ({
      ...props,
      [actionKey]: jest.fn(
        (...args) => dispatch(actionsObj[actionKey].apply(undefined, args))
      )
    }),
    {}
  )
)

const createMockMapDispatchToProps = (mapDispatchToProps) => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps
  }

  const mappingFunction = typeof mapDispatchToProps === 'function'
    ? mapDispatchToProps
    : getDispatchFunctionFormShorthandObject(mapDispatchToProps)

  return (dispatch, ownProps) => {
    dispatch = dispatch || mockDispatch

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    }
  }
}

const createMockMergeProps = (mergeProps) => {
  if (!mergeProps) {
    return mergeProps
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: mockDispatch }

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    }
  }
}

const mockReactRedux = {
  connect: (mapStateToProps, mapDispatchToProps, mergeProps) => (WrappedComponent) => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    WrappedComponent
  }),
  batch: (fn) => Promise.resolve(fn),
  useDispatch: jest.fn(() => mockDispatch),
  useSelector: jest.fn((selector) => selector())
}

export {
  mockReactRedux
}
