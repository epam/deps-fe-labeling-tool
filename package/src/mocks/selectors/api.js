const isSavingSelector = jest.fn(() => false)

const isRecognizingSelector = jest.fn(() => false)

const mockApiSelectors = {
  isRecognizingSelector,
  isSavingSelector
}

export {
  mockApiSelectors
}
