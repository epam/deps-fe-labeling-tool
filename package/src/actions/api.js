import { createAction } from 'redux-actions'
import { rotationAnglesSelector } from '@/selectors/canvas'
import { pageImageUrlSelector } from '@/selectors/document'
import { markupSelector } from '@/selectors/markup'
import { fieldsSelector } from '@/selectors/model'
import { primaryLanguageSelector } from '@/selectors/ocr'
import { getApi } from '@/services/api'

const FEATURE_NAME = 'API'

const requestAttempt = createAction(
  `${FEATURE_NAME}/REQUEST_ATTEMPT`
)

const requestSuccess = createAction(
  `${FEATURE_NAME}/REQUEST_SUCCESS`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

const requestFailure = createAction(
  `${FEATURE_NAME}/REQUEST_FAILURE`,
  (requestId, error) => ({
    requestId,
    error
  })
)

export const createRequestAction = (requestId, actionCreator) => {
  const requestActionCreator = (...args) => async (dispatch, getState) => {
    try {
      dispatch(requestAttempt(requestId))
      const result = await actionCreator(...args)(dispatch, getState)
      dispatch(requestSuccess(requestId))
      return result
    } catch (error) {
      dispatch(requestFailure(requestId, error.message))
      throw error
    }
  }
  requestActionCreator.toString = () => requestId
  return requestActionCreator
}

const save = createRequestAction(
  'save',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().save(markup, rotationAngles, language)
  }
)

const saveMarkup = createRequestAction(
  'saveMarkup',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    const fields = fieldsSelector(state)
    return getApi().saveMarkup(markup, rotationAngles, language, fields)
  }
)

const ocrText = createRequestAction(
  'ocrText',
  (engine, coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrText(
      engine,
      pageImageUrl,
      coordinates,
      language
    )
  }
)

const omrArea = createRequestAction(
  'omrArea',
  (coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)

    return getApi().omrArea(
      pageImageUrl,
      coordinates
    )
  }
)

const detectTables = createRequestAction(
  'detectTables',
  (selectedAreaCoords) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    return getApi().detectTables(
      pageImageUrl,
      selectedAreaCoords
    )
  }
)

const ocrTable = createRequestAction(
  'ocrTable',
  (engine, table) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrTable(
      engine,
      pageImageUrl,
      table,
      language
    )
  }
)

export {
  detectTables,
  ocrTable,
  ocrText,
  omrArea,
  resetDefault,
  requestAttempt,
  requestSuccess,
  requestFailure,
  save,
  saveMarkup
}
