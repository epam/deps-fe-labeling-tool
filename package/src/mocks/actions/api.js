import { mockMarkupSelectors } from '@/mocks/selectors/markup'

const omrArea = jest.fn(() => Promise.resolve({
  content: false,
  confidence: 0.77
}))

const ocrTable = jest.fn(() => Promise.resolve({ type: 'MOCK_OCR_TABLE' }))

const ocrText = jest.fn(() => Promise.resolve({
  content: 'MOCK_OCR_TEXT',
  confidence: 0.98
}))

const save = jest.fn(() => Promise.resolve(mockMarkupSelectors.markupSelector()))

const saveMarkup = jest.fn(() => Promise.resolve({ type: 'MOCK_SAVE_MARKUP' }))

const detectTables = jest.fn((url, coords) => Promise.resolve(['markupTable']))

const mockApiActions = {
  detectTables,
  omrArea,
  ocrTable,
  ocrText,
  save,
  saveMarkup
}

export {
  mockApiActions
}
