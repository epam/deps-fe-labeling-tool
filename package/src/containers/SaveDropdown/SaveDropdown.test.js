import { mockApiActions } from '@/mocks/actions/api'
import { mockMarkupActions } from '@/mocks/actions/markup'
import { mockReactRedux } from '@/mocks/mockReactRedux'
import { mockMarkupSelectors } from '@/mocks/selectors/markup'
import { mockModelSelectors } from '@/mocks/selectors/model'
import React from 'react'
import { shallow } from 'enzyme'
import { save, saveMarkup } from '@/actions/api'
import { updateInitialMarkup, storeAssignedMarkup } from '@/actions/markup'
import { getApi } from '@/services/api'
import { SaveDropdown } from '.'

const { WrappedComponent, mapStateToProps, mapDispatchToProps } = SaveDropdown

const SAVE_WITHOUT_EXTRACTION = 'Save without extraction'
const SAVE_AND_EXTRACT = 'Save & extract'
const SAVE_EXTRACT_AND_CLOSE = 'Save, extract & close'
const mockApiClose = jest.fn()
const mockSaveMarkup = jest.fn()
const mockSave = jest.fn()

const mockCancelAutoSave = jest.fn()

jest.mock('./useAutoSave', () => ({
  useAutoSave: jest.fn(() => mockCancelAutoSave)
}))

jest.mock('@/services/api', () => ({
  getApi: jest.fn(() => ({
    close: mockApiClose,
    saveMarkup: mockSaveMarkup,
    save: mockSave
  }))
}))

jest.mock('@/hocs/withHotKeys', () => ({
  withHotKeys: (Component) => Component
}))

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn((f) => f())
}))

jest.mock('react-redux', () => mockReactRedux)
jest.mock('@/actions/api', () => mockApiActions)
jest.mock('@/actions/markup', () => mockMarkupActions)
jest.mock('@/selectors/markup', () => mockMarkupSelectors)
jest.mock('@/selectors/model', () => mockModelSelectors)

describe('Container: SaveDropdown', () => {
  describe('mapStateToProps', () => {
    it('should pass expected state as props to WrappedComponent', () => {
      const { props } = mapStateToProps()
      expect(props.modifiedObjects).toEqual([])
    })
  })

  describe('mapDispatchToProps', () => {
    const { props } = mapDispatchToProps()

    it('should pass save action as prop to WrappedComponent', async () => {
      await props.save()
      expect(save).toHaveBeenCalledTimes(1)
    })

    it('should pass saveMarkup action as prop to WrappedComponent', async () => {
      await props.saveMarkup()
      expect(saveMarkup).toHaveBeenCalledTimes(1)
    })

    it('should pass updateInitialMarkup action as prop to WrappedComponent', () => {
      props.updateInitialMarkup()
      expect(updateInitialMarkup).toHaveBeenCalledTimes(1)
    })

    it('should pass storeAssignedMarkup action as prop to WrappedComponent', () => {
      props.storeAssignedMarkup()
      expect(storeAssignedMarkup).toHaveBeenCalledTimes(1)
    })
  })

  describe('component', () => {
    let wrapper, defaultProps

    beforeEach(() => {
      defaultProps = {
        ...mapStateToProps().props,
        ...mapDispatchToProps().props,
        registerHandlers: jest.fn()
      }
      delete defaultProps.modifiedObjects

      wrapper = shallow(<WrappedComponent {...defaultProps} />)
    })

    it('should render layout correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call registerHandlers on mount', () => {
      expect(defaultProps.registerHandlers).toHaveBeenCalled()
    })

    it('should render layout correctly if getApi().save is undefined', () => {
      getApi.mockReturnValueOnce({})
      wrapper = shallow(<WrappedComponent {...defaultProps} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render active Dropdown correctly', () => {
      const DropdownProps = wrapper.props().overlay()
      const MenuComponent = shallow(<div>{DropdownProps}</div>)
      expect(MenuComponent).toMatchSnapshot()
    })

    it('should call save storeAssignedMarkup when clicked on "Save & extract" item', async () => {
      const DropdownProps = wrapper.props().overlay()
      const MenuComponent = shallow(<div>{DropdownProps}</div>)
      const mockItem = {
        key: SAVE_AND_EXTRACT
      }

      await MenuComponent.props().children.props.onClick(mockItem)
      expect(defaultProps.save).toHaveBeenCalled()
      expect(defaultProps.storeAssignedMarkup).nthCalledWith(1, mockMarkupSelectors.markupSelector())
    })

    it('should call save and close when clicked on "Save, extract & close" item', async () => {
      const DropdownProps = wrapper.props().overlay()
      const MenuComponent = shallow(<div>{DropdownProps}</div>)
      const mockItem = {
        key: SAVE_EXTRACT_AND_CLOSE
      }

      await MenuComponent.props().children.props.onClick(mockItem)
      expect(defaultProps.save).toHaveBeenCalled()
      expect(mockApiClose).toHaveBeenCalled()
    })

    it('should call saveMarkup when clicked on "Save without extraction" item', async () => {
      const DropdownProps = wrapper.props().overlay()
      const MenuComponent = shallow(<div>{DropdownProps}</div>)
      const mockItem = {
        key: SAVE_WITHOUT_EXTRACTION
      }

      await MenuComponent.props().children.props.onClick(mockItem)
      expect(defaultProps.saveMarkup).toHaveBeenCalled()
    })
  })
})
