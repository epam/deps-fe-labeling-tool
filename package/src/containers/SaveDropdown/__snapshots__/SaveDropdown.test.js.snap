// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: SaveDropdown component should render active Dropdown correctly 1`] = `
<div>
  <SaveDropdownstyles__StyledMenu
    onClick={[Function]}
    trigger="click"
  >
    <MenuItem
      key="Save without extraction"
    >
      Save without extraction
    </MenuItem>
  </SaveDropdownstyles__StyledMenu>
</div>
`;

exports[`Container: SaveDropdown component should render layout correctly 1`] = `
<Dropdown
  disabled={true}
  overlay={[Function]}
  placement="bottomLeft"
  trigger="click"
>
  <IconButton
    icon={
      <SaveDropdownstyles__Wrapper>
        <SaveIcon />
        <DropdownIcon />
      </SaveDropdownstyles__Wrapper>
    }
  />
</Dropdown>
`;

exports[`Container: SaveDropdown component should render layout correctly if getApi().save is undefined 1`] = `
<Tooltip
  title="Save file"
>
  <IconButton
    disabled={true}
    icon={<SaveIcon />}
    onClick={[Function]}
  />
</Tooltip>
`;
